common4j/src/main/com/microsoft/identity/common/java/net/UrlConnectionHttpClient.java

```diff
@@ -183,16 +173,18 @@ private static void recordHttpTelemetryEventEnd(final HttpResponse response) {
      * @param requestUrl     The recipient {@link URL}.
      * @param requestHeaders Headers used to send the http request.
      * @param requestContent Optional request body, if applicable.
+     * @param sslContext     an optional {@link SSLContext} object.
      * @return HttpResponse      The response for this request.
      * @throws IOException If an error is encountered while servicing this request.
      */
     @Override
     public HttpResponse method(@NonNull final HttpClient.HttpMethod httpMethod,
                                @NonNull final URL requestUrl,
                                @NonNull final Map<String, String> requestHeaders,
-                               final byte[] requestContent) throws IOException {
+                               final byte[] requestContent,
+                               final SSLContext sslContext) throws IOException {
```

Comment:
I've been having to do some research here.  This should be an attribute of the Client, not a per-request feature, probably.

Reply:
The socket(factory) is set as part of HttpsUrlConnection, so it should be per-request.

(For example,  we could potentially reuse the client for 2 DRS requests for different WPJ certs - that's 2 SSL contexts.)

Reply:
Hm.  OK, having to go read through this code.  Are we doing this in a way that would allow us to reuse the SSL context for a given session? Right now, it looks to me like we're doing this afresh every time even when we may already have had a connection.  In any event, it's better than what we have.

Reply:
@AdamBJohnsonx  If this is not provided, then we will fall back to the default one.
If it's provided, it's up to the caller to specify the same one.

I could also create our own SSLContext that returns a wrapped Factory (instead of creating a new one every time this is provided).